Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -983,40 +982,63 @@ void DiscusMultipleScatteringCorrection::convertToLogWorkspace(const std::shared
  * meant that the k value going into the scatter was always fixed and equal to the overall kinc
  * The approach here will cope with multiple scatters by calculating a sumQSS at multiple
  * kinc values. These will be interpolated as required later on
+ * @param matWSs List of workspaces related to the structure factor for each sample/env component
+ * @param specialKs A list of special k values that the QSQ integral will be calculated for to reduce amount of
+ * interpolation required later on
  */
 void DiscusMultipleScatteringCorrection::calculateQSQIntegralAsFunctionOfK(ComponentWorkspaceMappings &matWSs,
                                                                            const std::vector<double> &specialKs) {
   for (auto &SQWSMapping : matWSs) {
     std::set<double> kValues(specialKs.begin(), specialKs.end());
```

Comment:
I think when you `push_back` it makes a copy of the element so for the elastic code it looks like you don't need to make a copy of the `specialKs` vector you only need to do that for the inelesitc code because you want to append more values. Is that right? Probably not a big deal anyway as `specialKs` probably isn't that big a vector (sort of the point of it!)